src/provider/provider_coarse.c

```diff
@@ -438,12 +717,12 @@ static enum umf_result_t coarse_memory_provider_alloc(void *provider,
         return UMF_RESULT_ERROR_MEMORY_PROVIDER_SPECIFIC;
     }
 
-    // Try to reuse existing blocks first.
-    // If the block that we want to reuse is has greater size, split it.
+    // Find first blocks with greater or equal size.
+    // If the block that we want to reuse has greater size, split it.
     // Try to merge split part with the successor if it is not used.
-    block_t *curr = coarse_provider->block_list;
-    while (curr) {
-        if (!curr->used && (curr->size > size)) {
+    block_t *curr = free_blocks_get_ge(coarse_provider->free_blocks, size);
+    if (curr) {
+        if ((curr->size > size)) {
```

Comment:
could you merge these two if() to get a less indented code?

Reply:
Done